frontend/src/components/pages/teacher/DisplayAssessmentResultsPage/index.tsx

```diff
@@ -43,24 +46,37 @@ const TAB_CONFIG = [
   },
 ];
 
-const isValidLocationState = (
+const getLocationState = (
   state: unknown,
-): state is { returnTo: string } => {
-  return typeof state === "object" && state !== null && "returnTo" in state;
-};
+): { returnTo?: string; sessionTitle?: string } => ({
+  returnTo: undefined,
+  sessionTitle: undefined,
+  ...(typeof state === "object" ? state : {}),
+});
 
 const DisplayAssessmentResults = () => {
   const { state } = useLocation();
-  const returnTo = isValidLocationState(state) ? state.returnTo : undefined;
+  const { returnTo, sessionTitle } = getLocationState(state);
+
+  // We need to fetch the test session title if we don't have it yet.
+  // This could happen if the user navigates directly to this page rather
+  // than from the test session list.
```

Comment:
instead of getting the test session title from the previous page, why don't we just return it when we query for student results? we have to hydrate the test associated with the test session anyway, so why not just return the title in that call? seems like it might simplify some logic here

let me know if im missing something though (:

Reply:
Doing it this way makes it so people can also manually construct URLs. Like if a teacher knows the ID of a class and wants to go there directly. An edge case for sure, but possible.

Reply:
gotcha